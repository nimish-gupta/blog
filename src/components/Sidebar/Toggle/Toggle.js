// @flow strict
import React from 'react';
import styled from 'styled-components';
import { IoMdSunny, IoMdMoon } from 'react-icons/io';

import { THEME } from '../../../constants';
import { useThemeContext } from '../../../context/theme-context';
import {
	marginBottom,
	marginTop,
} from '../../../assets/styled-components/mixins';

const ToggleContainer = styled.button`
	display: flex;
	justify-content: space-between;
	width: 69px;
	padding: 0.5rem;
	overflow: hidden;
	cursor: pointer;
	${marginBottom(1)};
	${marginTop(1)};
	font-size: ${(props) => props.theme.typographic.baseFontSize};
	border-radius: ${(props) => props.theme.button.borderRadius};
	background: ${(props) => props.theme.color.background};
	height: ${(props) => props.theme.button.height};
	border: 1px solid ${(props) => props.theme.color.grayBorder};
	color: ${(props) => props.theme.color.base};

	svg {
		width: 2.5rem;
		height: auto;
		transition: all 0.3s linear;

		&:nth-child(2) {
			transform: ${(props) =>
				`${props.isLightTheme ? 'translateY(0)' : 'translateY(100px)'}`};
		}

		&:first-child {
			transform: ${(props) =>
				`${props.isLightTheme ? 'translateY(-100px)' : 'translateY(0)'}`};
		}
	}
`;

type Props = {
	className?: string,
};

const Toggle = (props: Props) => {
	const { theme, toggleTheme } = useThemeContext();
	return (
		<ToggleContainer
			className={props.className}
			isLightTheme={theme === THEME.Light}
			onClick={toggleTheme}>
			<IoMdMoon />
			<IoMdSunny />
		</ToggleContainer>
	);
};
export default Toggle;
