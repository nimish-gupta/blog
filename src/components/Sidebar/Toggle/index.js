// @flow strict
export { default } from './Toggle';
