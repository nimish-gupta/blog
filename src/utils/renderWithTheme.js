import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import getTheme from '../assets/styled-components/_variables';

export default function renderWithTheme(component) {
	return renderer.create(
		<ThemeProvider theme={getTheme('light')}>{component}</ThemeProvider>
	);
}
