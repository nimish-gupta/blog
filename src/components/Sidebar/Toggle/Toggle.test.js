// @flow strict
import React from 'react';
import renderWithTheme from '../../../utils/renderWithTheme';
import Toggle from './Toggle';

describe('Toggle', () => {
	it('renders correctly', () => {
		const tree = renderWithTheme(<Toggle />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
