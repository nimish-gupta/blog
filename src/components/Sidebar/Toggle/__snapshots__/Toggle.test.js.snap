// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders correctly 1`] = `
<button
  className="sc-AxjAm kNKyhO"
  onClick={[Function]}
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M195 125c0-26.3 5.3-51.3 14.9-74.1C118.7 73 51 155.1 51 253c0 114.8 93.2 208 208 208 97.9 0 180-67.7 202.1-158.9-22.8 9.6-47.9 14.9-74.1 14.9-106 0-192-86-192-192z"
    />
  </svg>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M277.3 32h-42.7v64h42.7V32zm129.1 43.7L368 114.1l29.9 29.9 38.4-38.4-29.9-29.9zm-300.8 0l-29.9 29.9 38.4 38.4 29.9-29.9-38.4-38.4zM256 128c-70.4 0-128 57.6-128 128s57.6 128 128 128 128-57.6 128-128-57.6-128-128-128zm224 106.7h-64v42.7h64v-42.7zm-384 0H32v42.7h64v-42.7zM397.9 368L368 397.9l38.4 38.4 29.9-29.9-38.4-38.4zm-283.8 0l-38.4 38.4 29.9 29.9 38.4-38.4-29.9-29.9zm163.2 48h-42.7v64h42.7v-64z"
    />
  </svg>
</button>
`;
